### PR TITLE
feat: [CDS-77873]: Handling missing null checks for pipeline-service : NodeExecutionServiceImpl

### PR DESCRIPTION
**Pull request by Aninditaa-ch**
## Describe your changes
**Issue:** "nodeExecution" was nullable, NullPointerException could be added.

**Changes:** Handled the null check by adding if condition for nodeExecution and making some changes accordingly.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53812)

### CODE REVIEW

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger uts

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java [hunk 1]
```diff
@@ -831,13 +834,16 @@ void emitEvent(NodeExecution nodeExecution, OrchestrationEventType orchestration
       triggerPayload = metadata.getTriggerPayload() != null ? metadata.getTriggerPayload() : triggerPayload;
     }
 
-    Builder eventBuilder = OrchestrationEvent.newBuilder()
```
Comment (ankittiwari-harness) [inline pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java:834]: nit: why do we need to change this. As eventBuilder definition is still the same.

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java [hunk 2]
```diff
@@ -821,8 +821,11 @@ CloseableIterator<NodeExecution> fetchNodeExecutionsFromAnalytics(
 
   @VisibleForTesting
   void emitEvent(NodeExecution nodeExecution, OrchestrationEventType orchestrationEventType) {
+    if (nodeExecution == null){
```
Comment (ankittiwari-harness) [inline pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java:824]: nit: run git clang-format on the file.

**Comment by ankittiwari-harness** [comment]
Can you add description of what was the issue and how this fixes it

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by Aninditaa-ch — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53812)* status: 1 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java` line 0 at r2](https://reviewable.io/reviews/harness/harness-core/53812#gh-1349893811:-NgwIVDt63bMpO9npRan:bgmxbiu) ([raw file](https://github.com/harness/harness-core/blob/2778d7093313e09f1dfae03068715986f24fb86b/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java#L0)):*
<details><summary><i>Previously, ankittiwari-harness (Ankit Tiwari) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/harness/harness-core/pull/53812#discussion_r1349893811) to the right spot — sorry!)*

nit: why do we need to change this. As eventBuilder definition is still the same. 
</blockquote></details>

yeah right, I've removed it
___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java` line 824 at r2](https://reviewable.io/reviews/harness/harness-core/53812#gh-1349893995:-Ngw4BnA-nXH97ZBp3x2:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/2778d7093313e09f1dfae03068715986f24fb86b/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/executions/node/NodeExecutionServiceImpl.java#L824)):*
<details><summary><i>Previously, ankittiwari-harness (Ankit Tiwari) wrote…</i></summary><blockquote>

nit: run git clang-format on the file.
</blockquote></details>

done

**Review by shalini-agr — APPROVED** [review]

**Review by sahilhindwani-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53812)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53812)* status: 1 file(s) to review. 1. unresolved discussions.

**Comment by Aninditaa-ch** [comment]
trigger review

**Pull request merged**